src/corehost/common/pal.windows.cpp

```diff
@@ -464,8 +459,10 @@ bool pal::get_module_path(dll_t mod, string_t* recv)
 
 bool pal::get_temp_directory(pal::string_t& tmp_dir)
 {
-	pal::getenv(_X("TEMP"), &tmp_dir);
-	return pal::realpath(&tmp_dir);
+    pal::char_t temp_path[MAX_PATH + 1];
+    GetTempPathW(MAX_PATH + 1, temp_path);
```

Comment:
Check the result and return false - it's better than to rely on the realpath below - the `temp_path` is uninitialized, so if the `GetTempPath` fails, it will likely leave it uninitialized as well.